pycvvdp/display_model.py

```diff
@@ -308,6 +331,11 @@ def forward( self, V ):
             L = pq2lin( V ).clip(0.005, self.Y_peak) + Y_black + Y_refl #TODO: soft clipping
         elif self.EOTF=='linear':
             L = V.clip(max(0.005, Y_black), self.Y_peak) + Y_refl #TODO: soft clipping
+        elif self.EOTF=='HLG':
+            gamma = 1.2
+            if self.Y_peak > 1000:
+                gamma = 1.2 + 0.42 * math.log10(self.Y_peak / 1000) - 0.07623 * math.log10(self.E_ambient / 5)
+            L = (self.Y_peak-Y_black)*torch.pow(hlg2lin(V), gamma) + Y_black + Y_refl
```

Comment:
This implementation with this gamma is only correct when the EOTF is applied to the luma channel (Y). If we apply it to R, G, or B, it should be :

`L = (self.Y_peak-Y_black)*torch.pow(hlg2lin(Y), gamma - 1.0)*hlg2lin(V) + Y_black + Y_refl`

where Y = 0.2627 * R + 0.6780 * G + 0.0593 * B.

Note that if V = Y, the multiplication becomes `torch.pow(hlg2lin(Y), gamma - 1.0)*hlg2lin(Y)` which can be rewritten as `torch.pow(hlg2lin(Y), gamma - 1.0 + 1.0)`

Reply:
after doing some digging I confirmed that by the time we enter the forward functions we are in the RGB domain, therefore I implemented the RGB formulation of the EOTF, using the `Y**(gamma-1) * rgb`.

Reply:
LGTM! One nit -- would it be worth adding a check for the input being 3-dimensional (RGB) to make sure nobody tries to use it per channel? I am not sure if the code is permissive of such usage but I always prefer to reinforce the implementation. Just a soft suggestion.